### ISSUE TITLE
Add "flags: {}" to default Elm.Main.init in index.html

### ISSUE THREAD

**Issue by bburdette** [elm/compiler#1821]
**TLDR:**  when elm make src/main.elm generates index.html, have it add in a default flags:{} argument.  Then users who wish to use Browser.Document to make a single page with a title can do so by handling an empty flags arg in their init function.  If the user chooses Browser.sandbox, the extra flags arg is just ignored.

**Motivation:**

I want to make a simple self contained one page app - all code in index.html.  I'd like to set the title, so that means using Browser.Document.  
The problem is that document requires an init function that takes a flag: argument.  Index.html does not supply this. 
I have the option of editing in an argument in the generated index.html after every compile, or going to a separate index.html and main.js, which I don't want to do for this no-server page.

**Comment by rlefevre** [comment]
This is not needed, just use `()` as your flags type.
For example:
```
module Main exposing (main)

import Browser
import Html exposing (Html, text)


type alias Model =
    {}


main : Program () Model msg
main =
    Browser.document
        { init = always ( {}, Cmd.none )
        , subscriptions = always Sub.none
        , update = \msg model -> ( model, Cmd.none )
        , view = always { title = "title", body = [ text "hello" ] }
        }
```
You can compile it to index.html and test it.

**Comment by rlefevre** [comment]
I agree that it would be nice if it was explicitly described in the [flags section](https://guide.elm-lang.org/interop/flags.html) for example.
